### PR TITLE
Update user guide

### PR DESCRIPTION
**Pull request by logical-1985516**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -46,7 +46,33 @@ HireHub is a **desktop app for managing candidates, optimized for use via a Comm
...
+      - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.
+  * COUNTRY: must be a valid ISO-3166-1 alpha-2 code which can be found from https://www.iso.org/obp/ui/#search/code/.
+  * TAG: cannot be blank (except in edit command), and only alphanumeric characters and spaces are allowed.
+  * COMMENT: can be blank and does not have any constraints.
+  * TITLE: cannot be blank and has a character limit of 100.
+  * DESCRIPTION: can be blank and does not have any constraints.
+  * VACANCY: must be a positive integer.
+  * STATUS: must be one of the following 5 statuses (not case-sensitive): PRESCREEN, IN_PROGRESS, WAITLIST, OFFERED, REJECTED
+
+* These are the primary key (i.e. no 2 items can have the same parameter) of candidates, jobs and applications:
```
Comment (chiralcentre) [inline docs/UserGuide.md:72]: Add in the word "respectively" at the end

File: docs/UserGuide.md [hunk 2]
```diff
@@ -46,7 +46,33 @@ HireHub is a **desktop app for managing candidates, optimized for use via a Comm
...
+    * Other than the above constraints, there is no other validation to check if a country code is valid. Furthermore, given a country code, there is no phone validation specific to the country code. This would be part of the planned improvements.
+  * EMAIL: must be of the format local-part@domain and adhere to the following constraints:
+    * The local-part should only contain alphanumeric characters and these special characters, excluding the parentheses, (+_.-). The local-part may not start or end with any special characters.
+    * This is followed by a '@' and then a domain name. The domain name is made up of domain labels separated by periods. The domain name must:
+      - end with a domain label at least 2 characters long
+      - have each domain label start and end with alphanumeric characters
+      - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.
+  * COUNTRY: must be a valid ISO-3166-1 alpha-2 code which can be found from https://www.iso.org/obp/ui/#search/code/.
```
Comment (chiralcentre) [inline docs/UserGuide.md:64]: Mention that COUNTRY is case sensitive and must be in capital letters

**Review by chiralcentre — APPROVED** [review]
LGTM

File: docs/UserGuide.md (hunk 2 above)
Comment (logical-1985516) [reply docs/UserGuide.md]: Nice catch! Fixed

File: docs/UserGuide.md (hunk 1 above)
Comment (logical-1985516) [reply docs/UserGuide.md]: Fixed

**Pull request merged**